cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java

```diff
@@ -599,78 +599,30 @@ public DirectoryMerger getWorkspaceMerger() {
     }
-    ProcessContext pc = newProcess().directory(target).executable("git").withEnvVar("GIT_TERMINAL_PROMPT", "0");
-    if (Files.isDirectory(target.resolve(".git"))) {
-      ProcessResult result = pc.addArg("remote").run(true);
-      List<String> remotes = result.getOut();
-      if (remotes.isEmpty()) {
-        String message = "This is a local git repo with no remote - if you did this for testing, you may continue...\n"
-            + "Do you want to ignore the problem and continue anyhow?";
-        askToContinue(message);
-      } else {
-        pc.errorHandling(ProcessErrorHandling.WARNING);
-        result = pc.addArg("pull").run(false);
-        if (!result.isSuccessful()) {
-          String message = "Failed to update git repository at " + target;
-          if (this.offlineMode) {
-            warning(message);
-            interaction("Continuing as we are in offline mode - results may be outdated!");
-          } else {
-            error(message);
-            if (isOnline()) {
-              error("See above error for details. If you have local changes, please stash or revert and retry.");
-            } else {
-              error(
-                  "It seems you are offline - please ensure Internet connectivity and retry or activate offline mode (-o or --offline).");
-            }
-            askToContinue("Typically you should abort and fix the problem. Do you want to continue anyways?");
-          }
-        }
-      }
-    } else {
-      String branch = null;
-      int hashIndex = gitRepoUrl.indexOf("#");
-      if (hashIndex != -1) {
-        branch = gitRepoUrl.substring(hashIndex + 1);
-        gitRepoUrl = gitRepoUrl.substring(0, hashIndex);
-      }
-      this.fileAccess.mkdirs(target);
-      requireOnline("git clone of " + gitRepoUrl);
-      pc.addArg("clone");
-      if (isQuietMode()) {
-        pc.addArg("-q");
-      } else {
-      }
-      pc.addArgs("--recursive", gitRepoUrl, "--config", "core.autocrlf=false", ".");
-      pc.run();
-      if (branch != null) {
-        pc.addArgs("checkout", branch);
-        pc.run();
-      }
-    }
+    GitUtils gitUtils = new GitUtils(this, target, "origin", "master");
+    gitUtils.runGitPullOrClone(force, gitRepoUrl);
   }
 
   /**
-   * Checks if the Git repository in the specified target folder needs an update by
-   * inspecting the modification time of a magic file.
+   * Checks if the Git repository in the specified target folder needs an update by inspecting the modification time of
+   * a magic file.
    *
    * @param urlsPath The Path to the Urls repository.
    * @param repoUrl The git remote URL of the Urls repository.
+   * @param force boolean true enforces a git hard reset and cleanup of added files.
    */
-
-  private void gitPullOrCloneIfNeeded(Path urlsPath, String repoUrl) {
+  private void gitPullOrCloneIfNeeded(Path urlsPath, String repoUrl, boolean force) {
```

Comment:
great that you are following `SoC` and extracting git-logic out of `IdeContext` into its own git helper component.
But to do this consistent also this method needs to go there (making it `public`).

Reply:
Adjusted.